src/nvim/fileio.c

```diff
@@ -4876,35 +4874,39 @@ buf_check_timestamp (
 
   if (mesg != NULL) {
     path = home_replace_save(buf, buf->b_fname);
-    if (!helpmesg)
+    if (!helpmesg) {
       mesg2 = "";
-    tbuf = xmalloc(STRLEN(path) + STRLEN(mesg) + STRLEN(mesg2) + 2);
-    sprintf((char *)tbuf, mesg, path);
-    /* Set warningmsg here, before the unimportant and output-specific
-     * mesg2 has been appended. */
+    }
+    const size_t tbuf_len = STRLEN(path) + STRLEN(mesg) + STRLEN(mesg2) + 2;
+    char *const tbuf = xmalloc(tbuf_len);
+    snprintf(tbuf, tbuf_len, mesg, path);
+    // Set warningmsg here, before the unimportant and output-specific
+    // mesg2 has been appended.
     set_vim_var_string(VV_WARNINGMSG, tbuf, -1);
     if (can_reload) {
       if (*mesg2 != NUL) {
-        STRCAT(tbuf, "\n");
-        STRCAT(tbuf, mesg2);
+        strncat(tbuf, "\n", tbuf_len);
```

Comment:
That looks like a wrong use of the length argument of `strncat`.  (`tbuf_len` was previously computed big enough with "some" strings in mind, so it should work none the less.)

We use at most `tbuf_len` bytes from `src`, but if `strlen(tbuf) > 0` we could write over the end of `tbuf`.